third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -867,15 +876,58 @@ impl<'env> Generator<'env> {
             .map(|(e, t)| self.maybe_convert(e, &t))
             .collect::<Vec<_>>();
         let args = self.gen_arg_list(&args, true);
-        self.emit_with(id, |attr| {
-            Bytecode::Call(
-                attr,
-                targets,
-                BytecodeOperation::Function(fun.module_id, fun.id, type_args),
-                args,
-                None,
-            )
-        })
+        let ret_type: Type = self.env().get_function(fun).get_result_type();
+        // For the code such as `let x: &u64; let y: &u64; (x,y) = f() where f returns (&mut, &mut)`
+        // we need to generate a new temp with mut ref type and then freeze it
+        if ret_type.is_tuple() {
+            let mut new_targets = vec![];
+            let mut target_map = BTreeMap::new();
+            if let Type::Tuple(ret_tys) = ret_type {
```

Comment:
Move this up to eliminate the missing else branch.  (As mentioned above)

Reply:
done